perl.h

```diff
@@ -737,110 +737,156 @@ as in
+available.
 
-For a concrete example, see L<https://perlmonks.org/?node_id=11144355>.
+But, the declaration/initialization can be skipped if both the following are
+true:
 
-=item *
+=over
 
-Pass in the type for any declarations if it isn't known at macro definition
-time.  Suppose that the parameters to C<my_best_shot> in the example above
-didn't have to be UV.  This can be handled by changing the signature, something
-like this:
+=item 1.
 
- #define my_best_shot(a, a_type, b, b_type) STMT_START {    \
-                STMT_START {                                \
-                    a_type my_best_shot_a = a;              \
-                    b_type my_best_shot_b = b;              \
-                                                            \
-                    Size_t ix;                              \
-                                                            \
-                    ...                                     \
+a parameter is used just once in the macro, or it is known it will never
+have side effects; and
+
+=item 2.
+
+there is just the single parameter, or there is no possibility that this
+parameter will appear as part of an expression in another parameter, or vice
+versa
+
+=back
+
+An example would be:
+
+ // 'oob' stands for 'out-of-bounds'
+ #define act_on_oob(sv, action)             \
+    STMT_START {                            \
+        SV sv__act_on_oob = sv;             \
+                                            \
+        if (SvUV(sv__act_on_oob > IV_MAX) { \
+            action;                         \
+        }                                   \
+    } STMT_END
+
+act_on_oob(sv, abort());
+act_on_oob(sv, return);
+
+Also, if the parameter does have a type, the macro signature can include an
+extra parameter giving the type.
+
+ #define warn_if_odd(n, type)                               \
+    STMT_START {                                            \
+        type n__warn_if_odd = n;                            \
                                                             \
-                } STMT_END
+        if (n__warn_if_odd % 2 != 0) {                      \
+            my_warn(aTHX_ "%d is odd\n", n__warn_if_odd);   \
+        }                                                   \
+    } STMT_END
 
- my_best_shot(a + b + ix, int, a - b - ix, long);
+ warn_if_odd(var, long);
 
-You are out of luck if the signature can't be changed to include the types.
-Best then is to document that the macro cannot safely be called with parameters
-with side effects.  And any internally-declared variables should have names
-that won't clash with any outside names.  C<ix> should then be
-C<my_best_shot_ix>, for example.
+Note that C<type> didn't have to have a disambiguating name.  It would still
+work even if somewhere in the calling change C<type> was defined to be
+something; or if the call to the macro passed C<type> as the first positional
+parameter.
```

Comment:
>It would still
>work even if somewhere in the calling change C<type> was defined to be
>something; or if the call to the macro passed C<type> as the first positional
>parameter.

I don't understand either part of this sentence.